benches/gen_benchmark_inputs.py

```diff
@@ -67,17 +50,7 @@ def prepare_data(pos_no, cond_no):
 
         return krige_mat, k_vec, krige_cond
 
-    pos_no = 10000
-    cond_no = 500
-    krige_mat, k_vec, krige_cond = prepare_data(pos_no, cond_no)
-
-    np.savetxt(path / 'krige_vs_krige_mat.txt', krige_mat)
-    np.savetxt(path / 'krige_vs_k_vec.txt', k_vec)
-    np.savetxt(path / 'krige_vs_krige_cond.txt', krige_cond)
-
-    pos_no = 1000
-    cond_no = 50
-    krige_mat, k_vec, krige_cond = prepare_data(pos_no, cond_no)
```

Comment:
See above comment

Reply:
The question about typical problem sizes is hard to answer, as you might have already thought.

I just checked a [publication of mine](https://dx.doi.org/10.1016/j.advwatres.2016.06.012), where I used a very old implementation in C++ of the `summator_incompr` function. I used a 2d grid of 4600 x 1800 ~ 8*10^6 cells, which I would consider a larger problem size (for a 2d problem). I also actually used 6400 modes (len of `z1`, `z2`), which is probably a bit much (nowadays I always use exactly 1000 modes). Running one calcuation took 48 minutes on JURECA.

I guess @MuellerSeb can make much better estimates on typical sizes of kriging problems, as he has much more experience with it.

And just for the record, in this [publication](https://doi.org/10.1371/journal.pone.0254660), I estimated a variogram on an unstructured grid with 412 cells and a bin length of 28, which, computationally wise is rather small, but I think some variogram estimations are much smaller. 
In this [example](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/00_misc/04_herten.html#sphx-glr-examples-00-misc-04-herten-py), which uses non-synthetic data, I used 2000 data points and about 30 bins to estimate the variogram. But we actually only sampled a small part of the data due to computational costs.

But all in all, for small problem sizes, the run times are hardly noticeable, so I support the idea of using larger problem sizes and reducing the criterion sample sizes.

Regarding your final point to maybe only parallelize the outer most loops: as long as we are not working on 64 or more cores, that would probably at least be the more conservative solution and save some potential and surprising declines in performance.

Reply:
> would imply to drop the last commit, i.e. remove all parallelization except for the outermost loops.

Will wait for @MuellerSeb to chime in before making that change.